### PR TITLE
Add extension snapshot for oracle module to support state sync

### PR DESCRIPTION
**Pull request by RogerKSI**
## Implementation details
- Support state sync for the oracle module (sync os/ds in files folder in the snapshot)

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [x] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -586,6 +586,16 @@ func NewBandApp(
 	}
 	app.SetAnteHandler(anteHandler)
 	app.SetEndBlocker(app.EndBlocker)
+
+	if snapshotManager := app.SnapshotManager(); snapshotManager != nil {
```
Comment (colmazia) [inline app/app.go:590]: should add comment

**Review by colmazia — APPROVED** [review]
Let's Gamble, Try Merging

File: x/oracle/keeper/snapshotter.go [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+		}
+	}
+
+	return nil
+}
+
+func (os *OracleSnapshotter) Restore(
+	height uint64, format uint32, protoReader protoio.Reader,
+) (snapshot.SnapshotItem, error) {
+	if format == 1 {
```
Comment (taobun) [inline x/oracle/keeper/snapshotter.go:79]: SnapshotFormat

File: x/oracle/keeper/snapshotter.go [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+		compressedCode,
+		max(types.MaxExecutableSize, types.MaxWasmCodeSize, types.MaxCompiledWasmCodeSize),
+	)
+	if err != nil {
+		return sdkerrors.Wrapf(types.ErrUncompressionFailed, err.Error())
+	}
+
+	// check if we really need this file or not first
+	filename := filecache.GetFilename(code)
+	found, needed := foundCode[filename]
```
Comment (taobun) [inline x/oracle/keeper/snapshotter.go:172]: required

**Review by taobun — APPROVED** [review]

**Pull request merged**